src/components/ciConfig/TargetPlatformSelector.tsx

```diff
@@ -0,0 +1,162 @@
+import React from 'react'
+import { components } from 'react-select'
+import CreatableSelect from 'react-select/creatable'
+import { ReactComponent as WarningIcon } from '../../assets/icons/ic-warning.svg'
+import { TARGET_PLATFORM_LIST, tempMultiSelectStyles } from './CIConfig.utils'
+
+export interface TargetPlatformSelector {
+    allowOverride: boolean
+    selectedTargetPlatforms: any
+    setSelectedTargetPlatforms: any
+    showCustomPlatformWarning: boolean
+    setShowCustomPlatformWarning: (value: boolean) => void
+    targetPlatformMap: any
+    parentState?
+}
+
+function TargetPlatformSelector({
+    allowOverride,
+    selectedTargetPlatforms,
+    setSelectedTargetPlatforms,
+    showCustomPlatformWarning,
+    setShowCustomPlatformWarning,
+    targetPlatformMap,
+    parentState,
+}: TargetPlatformSelector) {
+    const platformMenuList = (props): JSX.Element => {
```

Comment:
extract `platformMenuList` out from this component as sometime this cause re renders

Reply:
doen